### ISSUE TITLE
Refactor EventEmitter to using store observer

### ISSUE THREAD

**Issue by louisgv** [concept-to-clinic/concept-to-clinic#302]
## Expected Behavior

With the store implemented, we should be able to use the value from it throughout the application and thus we can eliminate most EventEmitter implementation, since if there are communication cross component, it's best to do it once and use it everywhere.

## Current Behavior

At the moment, the following can be refactored to consume store state:

+ ImageSeries - TreeView EvenEmitter when selecting a new image

## Possible Solution

+ Compose a list of EventEmitter
+ Compose a list of store state that are nice to have
+ Refactor the EventEmitter into dispatcher and observer with the store.

## Checklist before submitting
- [ ] I have confirmed this using the officially supported Docker Compose setup using the `local.yml` configuration and ensured that I built the containers again and they reflect the most recent version of the project at the `HEAD` commit on the `master` branch 
- [ ] I have searched through the other currently open issues and am confident this is not a duplicate of an existing bug
- [ ] I provided a **minimal code snippet** or list of steps that reproduces the bug.
- [ ] I provided **screenshots** where appropriate
- [ ] I filled out all the relevant sections of this template

**Comment by louisgv** [comment]
After #288 I can start take a look at this. Those who have implemented EventEmitter please chime in.

**Comment by reubano** [comment]
Great idea!

**Comment by louisgv** [comment]
Since #288 is closed, I will start working on this issue.
